### PR TITLE
Improve skype detection

### PR DESCRIPTION
**Pull request by igorribeiroduarte**

### CODE REVIEW

File: src/lib/protocols/skype.c [hunk 1]
```diff
@@ -73,6 +103,21 @@ static void ndpi_check_skype(struct ndpi_detection_module_struct *ndpi_struct, s
 	    ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_SKYPE_CALL, NDPI_PROTOCOL_SKYPE);
 	  }
 	}
+
+        if (flow->detected_protocol_stack[0] == NDPI_PROTOCOL_UNKNOWN) {
+          const uint8_t id_flags_iv_crc_len = 11;
+          const uint8_t crc_len = 4;
```
Comment (utoni) [inline src/lib/protocols/skype.c]: I would prefer `sizeof(flow->l4.udp.skype_crc)` to document where this magic `4` comes from.

File: src/include/ndpi_typedefs.h [hunk 2]
```diff
@@ -775,6 +775,7 @@ struct ndpi_flow_udp_struct {
 
   /* NDPI_PROTOCOL_SKYPE */
   u_int8_t skype_packet_id;
+  char skype_crc[4];
```
Comment (utoni) [inline src/include/ndpi_typedefs.h]: An `u_int8_t` aka `unsigned char` would fit better as it indicates a non-printable byte buffer.

File: src/lib/protocols/skype.c (hunk 1 above)
Comment (igorribeiroduarte) [reply src/lib/protocols/skype.c]: Fixed.

File: src/include/ndpi_typedefs.h (hunk 2 above)
Comment (igorribeiroduarte) [reply src/include/ndpi_typedefs.h]: Fixed.

**Review by utoni — APPROVED** [review]

**Comment by lucaderi** [comment]
Thank you

**Pull request merged**